.github/workflows/ci.yml

```diff
@@ -111,7 +111,17 @@ jobs:
       - uses: julia-actions/julia-buildpkg@v1
         env:
           PYTHON: ""
-      - uses: julia-actions/julia-runtest@v1
+      - name: Run tests without coverage
+        uses: julia-actions/julia-runtest@v1
+        with:
+          coverage: false
+        env:
+          PYTHON: ""
+          TRIXI_TEST: ${{ matrix.trixi_test }}
+      - name: Run tests with coverage
+        uses: julia-actions/julia-runtest@v1
+        with:
+          coverage: true
```

Comment:
We also have quite some tests that do not use `maxiters`, e.g., https://github.com/trixi-framework/Trixi.jl/blob/main/test/test_unit.jl or the linear structure and jacobian stuff in https://github.com/trixi-framework/Trixi.jl/blob/main/test/test_special_elixirs.jl. How shall we handle them?

Reply:
Good question. For now, I think it shouldn't be a problem to run them twice, given that all tests are running in 20 minutes or less without coverage.

Reply:
The problem is that some of these tests are really expensive, e.g., the convergence tests. Maybe we could also make them cheaper when running with coverage?

Reply:
Ah, yeah. We should definitely do something about this.

Reply:
A general suggestion: What if we add testing kwargs `cells_per_dimension_coverage`, `initial_refinement_level_coverage`, and so on, to run coverage tests on the smallest possible mesh?

Reply:
Hmm, might be an idea. @sloede?

Reply:
Alternatively, coverage of convergence tests can probably run without further refinements. Coverage tests for visualization stuff can also run on a very small mesh etc.

Reply:
Or even better. You can pass any keyword followed by `_coverage` and we overwrite the default one if the tests are running with coverage. Then we could tweak all slow tests to be extremely fast for coverage.

Reply:
Or maybe adding a kwarg `coverage` where we can pass a `Dict` of kwargs to use with coverage tests would be a cleaner solution.

Reply:
I am not fully aware of the details anymore... are we talking about running a convergence test just to ensure test coverage (and that everything works in general) or are we actually running tests to ensure that some convergence numbers are maintained?

If we only run it to ensure coverage, I'd probably just duplicate the test(s) instead of cooking up a more elegant - but also more complex - scheme for this corner case.

Reply:
We run the full convergence test without coverage and compare EOCs. Then, we want to run a super simple convergence test again with coverage, assuming that the same code is covered as in the full test.

Reply:
As I said, I would try to use the simplest solution in the sense that if we are talking about only 2-4 tests that need special treatment, it probably is not worth the hassle to create an even more complex testing setup - instead, I'd just duplicate those specific tests.

However, if there is a clean solution like passing a dict or a named tuple as `coverage` to override arbitrary values, and if this would both be simple enough for regular tests and also cover the needs of the special tests, then this might be a good solution, too.